File: algorithms/[867]转置矩阵/solution.ts
// topics = ["数组"]

/**
 * Iterate
 * time O(mn), space O(1), m 和 n 分别为矩阵的行数和列数
 */
function transpose(matrix: number[][]): number[][] {
  const m = matrix.length;
  const n = matrix[0].length;

  const res = Array.from({ length: n }, () => new Array<number>(m));

  for (let i = 0; i < m; i += 1) {
    for (let j = 0; j < n; j += 1) {
      res[j][i] = matrix[i][j];
    }
  }

  return res;
}

File: algorithms/[867]转置矩阵/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        """
        Iterate
        time O(mn), space O(1), m 和 n 分别为矩阵的行数和列数
        """
        m, n = len(matrix), len(matrix[0])
        transposed = [[0] * m for _ in range(n)]
        for i in range(m):
            for j in range(n):
                transposed[j][i] = matrix[i][j]
        return transposed
